### ISSUE TITLE
Errors

### ISSUE THREAD

**Issue by snovak7** [confetti/yayson#20]
How do you render/trigger errors, when they happen? This is also in JSON API 1.0 specification.

**Comment by jede** [comment]
Right now this isn't handled by YAYSON. It would be great to get it in there though! If you would like to write a pull request that would be much appriciated :)

**Comment by snovak7** [comment]
Yes, there are then many options to consider, http response codes are involved here, so you need at least response object, which then can handle these, YAYSON doesn't even bother with content types as a quick browse through the code, but specification is quite explicit about it. But the most easy variant would be to "add errors", and then render would see if there are errors, then render error instead of the data, depends how deep you want to integrate it as a generic solution, or just as a quick serialisation to the user.

**Comment by jede** [comment]
Yeah, so YAYSON is built so you can transport your data as you wish. So we use it over websockets for instance. Maybe some kind of error presenter is the way to go?

**Comment by jlfama** [comment]
@snovak7 Did you ever work out a solution for this?

**Comment by jede** [comment]
How would you like to use an ErrorPresenter? Ive been thinking about adding one. Here is the full spec: http://jsonapi.org/format/#errors

I guess it makes sense that you inherit from the ErrorPresenter and there you specify the bout link for instance? And I guess the rest is just common attributes?

**Comment by jlfama** [comment]
For my use case, I ended up checking if the response is an error when I receive it. If it is an error I return the error, if it isn't an error, I run it through YAYSON.

```
  // Coffeescript, AngularJS
  responseHandler = (data) ->
    dataParsed = JSON.parse(data)
    if _.has(dataParsed, 'errors')
      return dataParsed
    Store = yayson().Store
    store = new Store()
    model = store.sync(dataParsed)
    return model
```
